src/Input/__tests__/Input.test.js

```diff
@@ -45,4 +51,90 @@ describe('<Input />', () => {
 
     expect(tree).toMatchSnapshot();
   });
+
+  context('with a validate function', () => {
+    const render = props =>
+      mount(
+        <ThemeProvider theme={theme}>
+          <Input
+            caption="The caption"
+            id="example"
+            validate={value => (value === '42' ? null : 'Value must be 42')}
+            value="42"
+            {...props}
+          />
+        </ThemeProvider>
+      );
+
+    it('displays the caption when valid', () => {
+      const wrapper = render();
+
+      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
+    });
+
+    it('displays the caption when valid and blurred', () => {
+      const wrapper = render();
+
+      wrapper.find(StyledInput).simulate('blur');
+
+      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
+    });
+
+    it('displays the caption when invalid but not yet blurred', () => {
+      const wrapper = render();
+      const input = wrapper.find(StyledInput);
+
+      input.simulate('change', { target: { value: '123' } });
+
+      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
+    });
+
+    it('displays the error message when invalid and blurred', () => {
+      const wrapper = render();
+      const input = wrapper.find(StyledInput);
+
+      input.simulate('change', { target: { value: '123' } });
+      input.simulate('blur');
+
+      expect(wrapper.find(FormField)).toHaveProp('caption', 'Value must be 42');
+    });
+
+    it('marks the input valid when valid', () => {
+      const wrapper = render();
+
+      expect(wrapper.find(StyledInput)).toHaveProp('isInvalid', false);
+    });
+
+    it('marks the input invalid when invalid', () => {
+      const wrapper = render();
+      const input = wrapper.find(StyledInput);
+
+      input.simulate('change', { target: { value: '123' } });
+      input.simulate('blur');
+
+      expect(wrapper.find(StyledInput)).toHaveProp('isInvalid', true);
+    });
+
+    it('calls onBlur with the original event', () => {
+      const onBlur = jest.fn();
+      const wrapper = render({ onBlur });
+
+      const fakeEvent = { id: 'fake-event' };
+
+      wrapper.find(StyledInput).simulate('blur', fakeEvent);
+
+      expect(onBlur).toHaveBeenCalledWith(expect.objectContaining(fakeEvent));
```

Comment:
Just doing `expect(onBlur).toHaveBeenCalledWith(fakeEvent)` here results in some sort of infinite loop and out of memory errors... 😕